Controller/Process/Redirect.php

```diff
@@ -155,108 +155,105 @@ public function execute()
-                    $order->addStatusHistoryComment(__('3D-secure validation was successful'))->save();
-                    // set back to false so when pressed back button on the success page
-                    // it will reactivate 3D secure
-                    $order->getPayment()->setAdditionalInformation('3dActive', '');
-                    $order->getPayment()->setAdditionalInformation('3dSuccess', true);
-
-                    if (!$this->_adyenHelper->isCreditCardVaultEnabled() &&
-                        !empty($result['additionalData']['recurring.recurringDetailReference'])) {
-                        $this->_adyenHelper->createAdyenBillingAgreement($order, $result['additionalData']);
-                    } elseif (!empty($result['additionalData']['recurring.recurringDetailReference'])
-                    ) {
-                        try {
-                            $additionalData = $result['additionalData'];
-                            $token = $additionalData['recurring.recurringDetailReference'];
-                            $expirationDate = $additionalData['expiryDate'];
-                            $cardType = $additionalData['paymentMethod'];
-                            $cardSummary = $additionalData['cardSummary'];
-                            /** @var PaymentTokenInterface $paymentToken */
-                            $paymentToken = $this->paymentTokenFactory->create(
-                                PaymentTokenFactoryInterface::TOKEN_TYPE_CREDIT_CARD
-                            );
-                            $paymentToken->setGatewayToken($token);
-                            $paymentToken->setExpiresAt($this->getExpirationDate($expirationDate));
-                            $details = [
-                                'type' => $cardType,
-                                'maskedCC' => $cardSummary,
-                                'expirationDate' => $expirationDate
-                            ];
-                            $paymentToken->setTokenDetails(json_encode($details));
-                            $extensionAttributes = $this->getExtensionAttributes($order->getPayment());
-                            $extensionAttributes->setVaultPaymentToken($paymentToken);
-                            $orderPayment = $order->getPayment()->setExtensionAttributes($extensionAttributes);
-                            $add = $this->serializer->unserialize($orderPayment->getAdditionalData());
-                            $add['force_save'] = true;
-                            $orderPayment->setAdditionalData($this->serializer->serialize($add));
-                            $this->orderPaymentResource->save($orderPayment);
-                        } catch (\Exception $e) {
-                            $this->_adyenLogger->error((string)$e->getMessage());
-                        }
+                $requestMD = $this->getRequest()->getPost('MD');
+                $requestPaRes = $this->getRequest()->getPost('PaRes');
```

Comment:
This might fix / has impact on #854 too.

Reply:
Same here we will investigate it in a separate issue